TypeCobol.LanguageServer/VsCodeProtocol/Signature Help/SignatureInformation.cs

```diff
@@ -41,14 +42,37 @@ public SignatureInformation(string label, string documentation, params Parameter
 
         public override bool Equals(object obj)
         {
-            if (!(obj is SignatureInformation))
-                return false;
+            return Equals(obj as SignatureInformation);
+        }
 
-            var givenSignature = (SignatureInformation) obj;
+        public bool Equals(SignatureInformation signatureInformation)
+        {
+            if (Object.ReferenceEquals(this, signatureInformation)) return true;
+            if (Object.ReferenceEquals(null, signatureInformation)) return false;
+
+            return signatureInformation.label == this.label
+                   && signatureInformation.documentation == this.documentation
+                   && CompareLists(this.parameters.ToList(), signatureInformation.parameters.ToList());
+        }
 
-            return givenSignature.label == this.label
-                   && givenSignature.documentation == this.documentation
-                   && CompareLists(this.parameters.ToList(), givenSignature.parameters.ToList());
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = 13;
+                hashCode = (hashCode * 397) ^ label.GetHashCode();
+                var docHashCode = documentation?.GetHashCode() ?? 0;
+                hashCode = (hashCode * 397) ^ docHashCode;
+                if (parameters != null)
+                {
+                    foreach (var param in parameters)
+                    {
+                        hashCode = (hashCode * 397) ^ param.GetHashCode();
```

Comment:
You should combine values from parameter itself because in the Equals method parameters are compared using their content (label and documentation) and here you are using the hashcode derived from the reference of the parameter object.